### PR TITLE
[IMP] stock, purchase_stock: update urls to the new format

### PR DESCRIPTION
**Pull request by sami-odoo**
This commit improves the readability of URLs by modifying them to a new, more human-friendly format. 
Additionally, the method `url_extract_rec_id_and_model` has been updated to enhance the extraction process for `model_name` and `rec_id`.

Task-3820230

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase_stock/tests/test_replenish_wizard.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def test_replenish_buy_1(self):
 
         last_po_id = False
         if purchase_order_id and model_name:
-            last_po_id = self.env[model_name[0]].browse(int(purchase_order_id[0]))
+            last_po_id = self.env[model_name].browse(int(purchase_order_id))
```
Comment (jpp-odoo) [inline addons/purchase_stock/tests/test_replenish_wizard.py:57]: Not related to the PR, should be in a different PR or commit

**Review by jpp-odoo — CHANGES_REQUESTED** [review]

File: addons/purchase_stock/tests/test_replenish_wizard.py (hunk 1 above)
Comment (sami-odoo) [reply addons/purchase_stock/tests/test_replenish_wizard.py:57]: Hello ,

https://github.com/odoo/odoo/blob/e428d7ae541b1a1f6cfd22116b946c880cf5132a/addons/purchase_stock/tests/test_replenish_wizard.py#L53

`def url_extract_rec_id_and_model` is extracting `model` and `rec_id` from the URL. And according to our current format of the URL's,  `action` is prioritised over `model` . So I first had to modify the function. And currently the modified function simply sends   model name and rec_id values and not list containing values. So I had to do some slight modifications in the test otherwise we would have a red build on the runbot.

Function Earlier:
https://github.com/odoo/odoo/blob/1fb5f4060eba52651e5d1503026ab3def3d358b7/addons/stock/tests/common.py#L115-L118 

if we want to avoid the diff we can send the `rec_id` and `model` by appending the values in separate lists

**Comment by jpp-odoo** [comment]
@robodoo r+
